### PR TITLE
Fix interaction of Ruin Abilities with themselves

### PR DESCRIPTION
**Pull request by HisuianZoroark**
https://www.smogon.com/forums/threads/bug-reports-v4-read-original-post-before-posting.3663703/page-104#post-9468014

### CODE REVIEW

File: data/abilities.ts [hunk 1]
```diff
@@ -357,7 +357,7 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
 		},
 		onAnyModifySpD(spd, target, source, move) {
 			const abilityHolder = this.effectState.target;
-			if (abilityHolder === target) return;
+			if (abilityHolder === target || target.hasAbility('Beads of Ruin')) return;
```
Comment (Karthik99999) [inline data/abilities.ts:360]: we can probably just shorten these to just the hasAbility check right? since that must return true on the ability holder

Comment (HisuianZoroark) [reply data/abilities.ts:360]: That feels correct

**Comment by HisuianZoroark** [comment]
Implemented all suggestions, thanks!

**Comment by DaWoblefet** [comment]
Thank you!

File: data/abilities.ts [hunk 2]
```diff
@@ -357,9 +357,9 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
 		},
 		onAnyModifySpD(spd, target, source, move) {
 			const abilityHolder = this.effectState.target;
-			if (abilityHolder === target) return;
+			if (target.hasAbility('Beads of Ruin')) return;
 			if (!move.ruinedSpD?.hasAbility('Beads of Ruin')) move.ruinedSpD = abilityHolder;
-			if (move.ruinedSpD !== abilityHolder && move.ruinedSpD !== target) return;
+			if (move.ruinedSpD !== abilityHolder) return;
```
Comment (urkerab) [inline data/abilities.ts:362]: (Although with the new mechanics couldn't `ruinedSpD` et. al. just be boolean?)

Comment (urkerab) [reply data/abilities.ts:362]: Oh, Def and SpD are harder because of spread moves. Never mind then.

**Pull request merged**